### DISCUSSION TITLE
Topics per class

### DISCUSSION THREAD

**Discussion by stewartjollymore** [MaartenGr/BERTopic#2215] [Ideas]
I think that topics per class has a lot of potential but I think that it would be even better if there was some logic to normalize the bar charts by the classes.  When one class dominates its usually bc there are just a lot of instances of it but does not really give a sense of a given topic is really "important" to that class.

I understand that it is not always the case and that this functionality would not be useful in all cases but adding some logic to calculate in that way could really extend the power of looking at topics per class.

**Comment by MaartenGr** [comment]
Thank you for addressing this. What kind of logic would you suggest that would be more appropriate for this specific instance?

**Reply by stewartjollymore** [reply]
The proportion of the topic by class to the entire class. Then you're comparing apples to apples.

E.g. two topics and class is sex (male, female) the bars for topic one would be the percent of males who discussed topic one.

**Reply by MaartenGr** [reply]
How would that implementation be different from the currently implemented `normalize_frequency` parameter?

**Reply by stewartjollymore** [reply]
Yeah that's what I was thinking. Since I do think there are times when it wouldn't be the right cut of the data but many times when it would be useful for comparing topics across the class levels.

Thanks for inquiring!! We love the product and use it for a lot of use cases!

**Reply by stewartjollymore** [reply]
Oh I see yup this is what I wanted :) thanks!
